ssc/cmod_pvwattsv7.cpp

```diff
@@ -0,0 +1,1261 @@
+					}
+					else
+					{
+						poa = 0;
+						tpoa = 0;
+						tmod = wf.tdry;
+						dc = 0;
+						ac = 0;
+					}
+
+					// transformer loss (night and day)
+					double iron_loss = pv.xfmr_nll_f * pv.xfmr_rating;
+					double winding_loss = pv.xfmr_ll_f * ac * (ac / pv.xfmr_rating);
+					double xfmr_loss = iron_loss + winding_loss;
+					if (y == 0) ld("ac_loss_transformer") += xfmr_loss * ts_hour; //ts_hour required to correctly convert to Wh for subhourly data
+					ac -= xfmr_loss;
+
+					p_stow[idx] = (tracker_stowing ? 1.0f : 0.0f);
+					p_shad_beam[idx] = (ssc_number_t)shad_beam; // might be updated by 1 axis self shading so report updated value
+
+					p_poa[idx] = (ssc_number_t)poa; // W/m2
+					p_tpoa[idx] = (ssc_number_t)tpoa;  // W/m2
+					p_tmod[idx] = (ssc_number_t)tmod;
+					p_dc[idx] = (ssc_number_t)dc; // power, Watts
+					p_ac[idx] = (ssc_number_t)ac; // power, Watts
+
+					// accumulate hourly energy (kWh) (was initialized to zero when allocated)
+					p_gen[idx_life] = (ssc_number_t)(ac * haf(hour) * util::watt_to_kilowatt * degradationFactor[y]);
+
+					if (y == 0) { //report first year annual energy
+						annual_kwh += p_gen[idx] / step_per_hour;
+					}
+
+					if (y == 0) ld("ac_loss_adjustments") += ac * (1.0 - haf(hour)) * ts_hour; //ts_hour required to correctly convert to Wh for subhourly data
+					if (y == 0) ld("ac_delivered") += ac * haf(hour) * ts_hour; //ts_hour required to correctly convert to Wh for subhourly data
+
+					idx++;
+					idx_life++;
+				}
+			}
+
+			wdprov->rewind();
+			if (y == 0) {
+				accumulate_monthly("gen", "monthly_energy", ts_hour);
+				accumulate_annual("gen", "annual_energy", ts_hour);
+			}
+		}
+
+		accumulate_monthly("dc", "dc_monthly", 0.001*ts_hour);
+		accumulate_monthly("ac", "ac_monthly", 0.001*ts_hour);
+
+		ssc_number_t *poam = accumulate_monthly("poa", "poa_monthly", 0.001*ts_hour); // convert to energy
+		ssc_number_t *solrad = allocate("solrad_monthly", 12);
+		ssc_number_t solrad_ann = 0;
+		for (int m = 0; m < 12; m++)
+		{
+			solrad[m] = poam[m] / util::nday[m];
+			solrad_ann += solrad[m];
+		}
+		assign("solrad_annual", var_data(solrad_ann / 12));
+
+		accumulate_annual("ac", "ac_annual", 0.001*ts_hour);
+
+		assign("location", var_data(hdr.location));
+		assign("city", var_data(hdr.city));
+		assign("state", var_data(hdr.state));
+		assign("lat", var_data((ssc_number_t)hdr.lat));
+		assign("lon", var_data((ssc_number_t)hdr.lon));
+		assign("tz", var_data((ssc_number_t)hdr.tz));
+		assign("elev", var_data((ssc_number_t)hdr.elev));
+		assign("percent_complete", var_data((ssc_number_t)percent));
+
+		double gcr_for_land = pv.gcr;
+		if (gcr_for_land < 0.01) gcr_for_land = 1.0;
+		double landf = is_assigned("landf") ? as_number("landf") : 1.0f;
+		assign("land_acres", var_data((ssc_number_t)(landf * module_m2 / gcr_for_land * 0.0002471)));
+
+		// for battery model, force inverter model
+		assign("inverter_model", var_data((ssc_number_t)4));
```

Comment:
It confusingly doesn't actually correspond to the inverter models in pvsamv1- it's a required input to battwatts. I think it had been defined in PVWattsV5 to avoid it being a separate required input in the UI, since when calling battwatts from PVWatts it should always be 4... but I personally think it makes more sense to remove it as an output of PVWatts and set the default value in BattWatts to 4. I'll do that and see if it works.

Reply:
yeah the inverter model is set to SHARED_INVERTER::NONE in battwatts so it doesn't actually make sense to have the inverter model as an input to battwatts

Reply:
It seems to be used to determine how to get inverter efficiency? Line 250

Reply:
https://github.com/NREL/ssc/blob/4cd757a1796f06c0addd057b4a9f379b961044f4/ssc/cmod_battwatts.cpp#L250

line 250 just makes sure inverter model is a reasonable number before assigning the efficiency. but if it isn't a reasonable number. if it's less than 4, then the inverter efficiency is unassigned...

I changed this code in resiliency branch to be simply:

```
    // Inverter model must be
    batt_vars->inverter_model = SharedInverter::NONE;
    batt_vars->inverter_efficiency = inv_eff;
```

in which case we don't really need the inverter_model

Reply:
I updated mine accordingly- although the SharedInverter reference didn't work for me, so it's hard-coded to 4 right now. I also updated the logic in cmod_battery.